scripts/check_branch.py

```diff
@@ -21,9 +24,37 @@
 log = logging.getLogger(__name__)
 
 
-def default_deployment(branch: str | None) -> str | None:
+class BranchContext(Enum):
```

Comment:
I think this deserves a docstring as it isn't immediately apparent (to me, at least) what a *branch context* is.